app/src/pages/playground/SpanPlaygroundPage.tsx

```diff
@@ -1,10 +1,31 @@
-import React from "react";
+import React, { useMemo } from "react";
+import { useLoaderData } from "react-router";
 
-// import { useLoaderData } from "react-router";
+import { spanPlaygroundPageLoaderQuery$data } from "./__generated__/spanPlaygroundPageLoaderQuery.graphql";
 import { Playground } from "./Playground";
+import { transformSpanAttributesToPlaygroundInstance } from "./utils";
 
 export function SpanPlaygroundPage() {
-  // const data = useLoaderData();
+  const data = useLoaderData() as spanPlaygroundPageLoaderQuery$data;
+  const span = useMemo(() => {
+    if (data.span.__typename === "Span") {
+      return data.span;
+    }
+    return null;
+  }, [data.span]);
 
-  return <Playground />;
+  if (!span) {
+    throw new Error("Invalid span");
```

Comment:
Can we avoid this with @required? If not, maybe throw a more user-friendly message

Reply:
It's actually not nullable, but above this we return null in the event that the `__typename` is not Span, this is because of relay adding other to the type to be safe, will update the error message but should never happen

```typescript
  // This will never be '%other', but we need some
    // value in case none of the concrete values match.
    readonly __typename: "%other";
```